### PR TITLE
Add ProductAppearance attribute support.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
This is a draft implementation of https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/6113

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml [hunk 1]
```diff
@@ -23,6 +23,44 @@ limitations under the License.
...
+    <item name="Gold" value="19"/>
+  </enum>
+
+  <struct name="ProductAppearanceStruct">
+    <cluster code="0x0028"/>
+    <item fieldId="0" name="Finish"       type="ProductFinishEnum" isNullable="false"/>
+    <item fieldId="1" name="PrimaryColor" type="ColorEnum"         isNullable="true" />
+  </struct>
+
   <cluster singleton="true">
```
Comment (tcarmelveilleux) [inline src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml:64]: You need to bump the revision as well.

Comment (bzbarsky-apple) [reply src/app/zap-templates/zcl/data-model/chip/basic-information-cluster.xml:67]: Good catch, done.

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/clusters/basic-information/basic-information.cpp [hunk 2]
```diff
@@ -343,6 +349,33 @@ CHIP_ERROR BasicAttrAccess::WriteLocation(AttributeValueDecoder & aDecoder)
...
+    auto * provider = GetDeviceInstanceInfoProvider();
+    ProductFinishEnum finish;
+    ReturnErrorOnFailure(provider->GetProductFinish(&finish));
+
+    ColorEnum color;
+    CHIP_ERROR colorStatus = provider->GetProductPrimaryColor(&color);
+    if (colorStatus != CHIP_NO_ERROR && colorStatus != CHIP_ERROR_NOT_IMPLEMENTED)
+    {
+        return colorStatus;
+    }
```
Comment (andy31415) [inline src/app/clusters/basic-information/basic-information.cpp:363]: if colorStatus is NOT_IMPLEMENTED, should we explicitly set a default?

**Review by andy31415 — APPROVED** [review]

File: src/app/clusters/basic-information/basic-information.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/clusters/basic-information/basic-information.cpp:363]: No; this field is optional in the spec.

**Pull request merged**